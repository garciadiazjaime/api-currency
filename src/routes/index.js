import RateController from '../controllers/rateController';
import UsdMxnController from '../controllers/usdmxnController';

function routes(server) {
  server.get('/rates', RateController.list);
  server.post('/rates', RateController.save);

  server.get('/rates/usdmxn', UsdMxnController.list);
  server.post('/rates/usdmxn', UsdMxnController.save);
}

export default routes;
