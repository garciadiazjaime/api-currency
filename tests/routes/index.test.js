import routes from '../../src/routes';

describe('routes', () => {
  it('makes expected calls', () => {
    const server = {
      get: jest.fn(),
      post: jest.fn(),
    };

    routes(server);

    expect(server.get.mock.calls).toMatchSnapshot();

    expect(server.post.mock.calls).toMatchSnapshot();
  });
});
