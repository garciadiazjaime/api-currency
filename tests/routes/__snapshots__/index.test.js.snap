// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes makes expected calls 1`] = `
Array [
  Array [
    "/rates",
    [Function],
  ],
  Array [
    "/rates/usdmxn",
    [Function],
  ],
]
`;

exports[`routes makes expected calls 2`] = `
Array [
  Array [
    "/rates",
    [Function],
  ],
  Array [
    "/rates/usdmxn",
    [Function],
  ],
]
`;
